sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/MergeRowsExec.scala

```diff
@@ -257,4 +272,27 @@ case class MergeRowsExec(
       null
     }
   }
+
+  // For group based merge, copy is inserted if row matches no other case
+  private def incrementCopyMetric(): Unit = longMetric("numTargetRowsCopied") += 1
```

Comment:
What's the cost of doing this per each row? I know we implement [some tricks for regular writes to update metrics only once per 100 rows](https://github.com/apache/spark/blob/4ec992c773968b78ab2eebfd4bd3a1d55d23dcb1/sql/core/src/main/scala/org/apache/spark/sql/execution/metric/CustomMetrics.scala#L26). Do we need to worry about it here?

Reply:
oh good point. We should do the same here.

Reply:
For that one, I took a closer look.  It look like it was from this discussion:  https://github.com/apache/spark/pull/31451#discussion_r604572473.  In that case there was concern to get the metric from the DSV2 connector, to avoid calling currentMetricsValue so many times because its heavy.

In our case, getting the metric is in memory, so it should be quick.

On the sending end, it looks like SQLMetric is an accumulator and updating it just sets an in memory value as well.

So at first glance, I think adding complexity to update the metric per 100 rows may not be worth it.  But I may be missing something.